src/hw/pvr/tr.c

```diff
@@ -597,6 +598,14 @@ static void tr_parse_poly_param(struct tr *tr, const struct tile_ctx *ctx,
     surf->depth_func = DEPTH_GEQUAL;
   }
 
+  /* pass polygon type to surface */
```

Comment:
I don't believe the poly type is needed for other effects at the moment.

With that in mind, I'd do the minimum work here and just add an `int pt_alpha_test` on the surf and do `surf->pt_alpha_test = tr->list_type == TA_LIST_PUNCH_THROUGH`.